Basic prototype

YuriyVaadin opened the pull request:
Basic functionality implemented based on checkbox element with needed improvements

Connected to https://github.com/vaadin/components-team-tasks/issues/260

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-radio-button/2)
<!-- Reviewable:end -->

web-padawan commented:
Review status: 0 of 7 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-radio-button.html, line 90 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-Knmsug8BPzXSkvgiHOx:-Knmsug8BPzXSkvgiHOy:b-dl9qpa) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L90)):*
> ```HTML
>       static get properties() {
>         return {
>           
> ```

nit: seems you have to configure your editor to remove trailing spaces. ESLint rule `no-trailing-spaces` could be useful here.

---

*[vaadin-radio-button.html, line 111 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmsFSXL3g8gL566BYY:-KnmsFSYyRCCZte1hB1V:bjn6btj) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L111)):*
> ```HTML
>             type: String,
>             reflectToAttribute: true
>           },
> ```

nit: trailing comma could be removed

---

*[vaadin-radio-button.html, line 115 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-Knmt_eeMzpS4hn1a8ob:-Knmt_eeMzpS4hn1a8oc:bwyzo) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L115)):*
> ```HTML
>         };
>       }
>       ready() {
> ```

nit: adding empty lines between methods usually improve readability 

---

*[vaadin-radio-group.html, line 79 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmtyoIj9FiuhLD4i6Y:-KnmtyoIj9FiuhLD4i6Z:bismaf1) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-group.html#L79)):*
> ```HTML
>             observer: '_disabledChanged'
>           }
>           
> ```

nit: same about trailing spaces here, and also consider adding empty lines between properties as well

---

*[vaadin-radio-group.html, line 139 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmuIEtnSOgSlb8CYl-:-KnmuIEtnSOgSlb8CYl0:bct4fgx) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-group.html#L139)):*
> ```HTML
>       }
>       validate() {
>         
> ```

nit: another bunch of trailing spaces here and below, and seems that git warns about `newline at end of file`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 0 of 21 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-radio-button-group.html, line 79 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmtyoIj9FiuhLD4i6Y:-KnnJa9kVfGls2TIxRYf:b-43znlp) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button-group.html#L79)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

nit: same about trailing spaces here, and also consider adding empty lines between properties as well
</blockquote></details>

Done in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/972067a337b10cfa3a3cafa432f68d9f9de53f2d

---

*[vaadin-radio-button-group.html, line 139 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmuIEtnSOgSlb8CYl-:-KnnJQ2WPtEBdYusamP2:b3qb278) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button-group.html#L139)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

nit: another bunch of trailing spaces here and below, and seems that git warns about `newline at end of file`
</blockquote></details>

Fixed and removed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/972067a337b10cfa3a3cafa432f68d9f9de53f2d

---

*[vaadin-radio-button.html, line 90 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-Knmsug8BPzXSkvgiHOx:-KnnK3Yn8z8Z1gnJMAZA:b-896fix) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L90)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

nit: seems you have to configure your editor to remove trailing spaces. ESLint rule `no-trailing-spaces` could be useful here.
</blockquote></details>

Done.

---

*[vaadin-radio-button.html, line 111 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnmsFSXL3g8gL566BYY:-KnnJgR092TuruqEOQmg:b8buifj) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L111)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

nit: trailing comma could be removed
</blockquote></details>

Removed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/972067a337b10cfa3a3cafa432f68d9f9de53f2d

---

*[vaadin-radio-button.html, line 115 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-Knmt_eeMzpS4hn1a8ob:-KnnJjNli_bWNTOmXku6:b4t4qlh) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1eed03d51b080549f20d4e5957b27cc934111cc8/vaadin-radio-button.html#L115)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

nit: adding empty lines between methods usually improve readability 
</blockquote></details>

 Added in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/972067a337b10cfa3a3cafa432f68d9f9de53f2d

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

limonte commented:
Reviewed 1 of 8 files at r1, 16 of 19 files at r2, 4 of 4 files at r4, 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[vaadin-radio-button-group.html, line 87 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnnhFlpNBM7gxIXhCgD:-KnnhFlpNBM7gxIXhCgE:b1bi4xf) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/e7af45c7f8a56678c308d95fd22e450c24d9301f/vaadin-radio-button-group.html#L87)):*
> ```HTML
>           return 'vaadin-radio-button-group';
>         }
>         static get properties() {
> ```

empty line before

---

*[demo/group.html, line 8 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnngugtXfmK1VZXNwdQ:-KnngugtXfmK1VZXNwdR:b-dselky) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/e7af45c7f8a56678c308d95fd22e450c24d9301f/demo/group.html#L8)):*
> ```HTML
>   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
> 
>   <title>vaadin-radio-button Basic Examples</title>
> ```

vaadin-radio-button-group Examples

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 21 of 23 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[vaadin-radio-button-group.html, line 87 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnnhFlpNBM7gxIXhCgD:-KnniVu4vjxRDqd6BKh9:bcxupuv) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/e7af45c7f8a56678c308d95fd22e450c24d9301f/vaadin-radio-button-group.html#L87)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

empty line before
</blockquote></details>

Done in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/7c70d77b71ca6cf002f5f3a8f8c596a3e46fb18b

---

*[demo/group.html, line 8 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KnngugtXfmK1VZXNwdQ:-KnniXhmeno9QcxJHjEO:bcxupuv) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/e7af45c7f8a56678c308d95fd22e450c24d9301f/demo/group.html#L8)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

vaadin-radio-button-group Examples
</blockquote></details>

Done in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/7c70d77b71ca6cf002f5f3a8f8c596a3e46fb18b

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

manolo commented:
You need to fix linter errors.  Run `polymer lint --rules polymer-2 --input *.html` before committing so as travis does not fail.

---

Review status: 20 of 23 files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-Ko65klKqeJGimKWYUvp:b-tva0sf)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
The linter errors are now fixed.

---

Review status: 19 of 23 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-Ko6tpGDlAeACwVzJ2Gg:b-var787)*
<!-- Sent from Reviewable.io -->

limonte commented:
Reviewed 1 of 19 files at r2, 2 of 2 files at r6.
Review status: 20 of 23 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-radio-button-group.html, line 117 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoBMpTbl-Fwe4yiDzhb:-KoBMpTbl-Fwe4yiDzhc:b83027i) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2a0b6f5ba8697031385a85f19ccdf8c516623da7/vaadin-radio-button-group.html#L117)):*
> ```HTML
>         _disabledChanged(disabled) {
>           if (disabled) {
>             this.removeAttribute('invalid');
> ```

The Single Responsibility Principle is violated here. There should be only one place for handling `invalid` attribute.

---

*[vaadin-radio-button-group.html, line 170 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoBMnx4mc3vjiwllslf:-KoBMnx5y5fW2Q1XMF2P:b83027i) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2a0b6f5ba8697031385a85f19ccdf8c516623da7/vaadin-radio-button-group.html#L170)):*
> ```HTML
>         _valueChanged(value) {
>           if (value) {
>             this.removeAttribute('invalid');
> ```

The Single Responsibility Principle is violated here. There should be only one place for handling `invalid` attribute.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 20 of 26 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-radio-button-group.html, line 117 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoBMpTbl-Fwe4yiDzhb:-KoBbCeEILMmbk-ktOBk:b-896fix) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2a0b6f5ba8697031385a85f19ccdf8c516623da7/vaadin-radio-button-group.html#L117)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

The Single Responsibility Principle is violated here. There should be only one place for handling `invalid` attribute.
</blockquote></details>

Done.

---

*[vaadin-radio-button-group.html, line 170 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoBMnx4mc3vjiwllslf:-KoBbGqJbuXqAD9q3y4L:b-896fix) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2a0b6f5ba8697031385a85f19ccdf8c516623da7/vaadin-radio-button-group.html#L170)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

The Single Responsibility Principle is violated here. There should be only one place for handling `invalid` attribute.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

platosha commented:
Missing some parts in accessibility, but it’s not required right now, can be improved after this PR is merged.

API docs are missing. This is better to do right now. Normally we avoid merging undocumented features.

---

Reviewed 1 of 19 files at r2, 1 of 1 files at r7, 1 of 2 files at r8, 3 of 4 files at r9, 1 of 1 files at r10, 2 of 2 files at r11, 4 of 4 files at r12.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks broke.

---

*[vaadin-radio-button-group.html, line 18 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCTqBZkX14k_gEeRBc:-KoCTqBZkX14k_gEeRBd:bvvhk6g) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L18)):*
> ```HTML
> 
>       [part="wrapper"] {
>         display: inline-flex;
> ```

Do we really need flex layout for the radio buttons? Why?

---

*[vaadin-radio-button-group.html, line 32 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCQMun5Oxo6bRVGhYK:-KoCQMun5Oxo6bRVGhYL:b-sr4zc9) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L32)):*
> ```HTML
> <dom-module id="vaadin-radio-button-group">
>   <template>
>     <div part="wrapper" id="wrapper">
> ```

Is wrapper necessary here? Why?

---

*[vaadin-radio-button-group.html, line 60 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCdkrTpoqby_nVDABI:-KoCdkrTpoqby_nVDABJ:b-n88xov) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L60)):*
> ```HTML
>        * Attribute    | Description
>        * -------------|------------
>        * `active`     | Set when the checkbox is pressed down, either with mouse, touch or the keyboard.
> ```

This entire styling doc section is copied from checkbox, seems not relevant to the radio button group.

Let’s remove it for now. Another option is to rewrite it, so that it describes relevant stuff here.

---

*[vaadin-radio-button-group.html, line 163 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCXgqOko-tca-8c8z7:-KoCXgqOko-tca-8c8z8:b1cgnpd) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L163)):*
> ```HTML
> 
>         _forEachButton(fn) {
>           Array.prototype.forEach.call(this.querySelectorAll('vaadin-radio-button'), fn);
> ```

`querySelectorAll` will also pick up any deep radio buttons. Not sure if it’s a good thing.

Perhaps, for the initial version, it’s better to restrict nesting and only allow direct child radio buttons to participate in the group. We can relax this restriction later, when a need comes.

Filtering from `children` seems more appropriate here instead of `querySelectorAll` then.

BTW, In ES6 you can do `Array.from(this.children).filter(...).forEach(...)`.

---

*[vaadin-radio-button.html, line 36 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCcpL44gFJkLqqPuC3:-KoCcpL5bfvDIgOrRVZ8:b-i85np6) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L36)):*
> ```HTML
>       [part="label"] {
>         padding: 0 5px;
>         outline: none;
> ```

Afaik span should not have a native outline, no need for this like then. Please check if it is necessary.

---

*[vaadin-radio-button.html, line 85 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCeyEfE4AwPqaWf7e6:-KoCeyEgXipPMVDJE5RY:b-iv68v9) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L85)):*
> ```HTML
>        * Part name         | Description
>        * ------------------|----------------
>        * `wrapper`         | The `<label>` element which wrapps the checkbox and [part="label"]
> ```

`checkbox` copy-pasting leftover here and below.

---

*[vaadin-radio-button.html, line 151 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCixd4V6wvPalFvXaS:-KoCixd4V6wvPalFvXaT:b-8s37b5) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L151)):*
> ```HTML
>           this.setAttribute('aria-checked', checked);
> 
>           this.dispatchEvent(new CustomEvent('radio-changed', {bubbles: true}));
> ```

Hm-m, I’m not completely sure, but seems that we can simply call this event as `'change'`. 

Interference with the native `change` should not be an issue in this case. The original event from the internal native radio is not composed, therefore it should not propagate outside the shadow root.

Could you, please, try?

---

*[test/vaadin-radio-button_test.html, line 24 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCl90Tb6H4lL4SDpcW:-KoCl90UrQ8wOOdv59r_:b8201nn) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/test/vaadin-radio-button_test.html#L24)):*
> ```HTML
> 
>   <script>
>     describe('vaadin-radio-button-group', () => {
> ```

Seems like some tests are for the group, but some are just for the radio button.

We usually have separate test files for separate element files. Maybe split this up a bit?

---

*[test/vaadin-radio-button_test.html, line 31 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoClgROMCZe5yO9i_wB:-KoClgRP2MS8tqLWqJJO:b-u2zl6o) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/test/vaadin-radio-button_test.html#L31)):*
> ```HTML
>         vaadinRadioButtonList = vaadinRadioButtonGroup.querySelectorAll('vaadin-radio-button');
>         nativeRadio = Polymer.dom(vaadinRadioButtonList[0].root).querySelector('[part="native-radio"]');
>         label = Polymer.dom(vaadinRadioButtonList[0].root).querySelector('[part="label"]');
> ```

Using Polymer.dom is unnecessary with Polymer 2. You should be able to use the DOM APIs directly.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoC_S6-e-AkdPa2C8B5:b-81ncku)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks broke.

---

*[vaadin-radio-button-group.html, line 163 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCXgqOko-tca-8c8z7:-KoDB10VJcN7rBfK8Tkg:b31y3xe) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L163)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

`querySelectorAll` will also pick up any deep radio buttons. Not sure if it’s a good thing.

Perhaps, for the initial version, it’s better to restrict nesting and only allow direct child radio buttons to participate in the group. We can relax this restriction later, when a need comes.

Filtering from `children` seems more appropriate here instead of `querySelectorAll` then.

BTW, In ES6 you can do `Array.from(this.children).filter(...).forEach(...)`.
</blockquote></details>

probably specifying a closer selector '> vaadin-radio-button'  works without having to chain methods

---

*[vaadin-radio-button.html, line 144 at r12](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoDBs1EXKqlqogNY607:-KoDBs1EXKqlqogNY608:b-3m7c20) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/88dfbcfd6cbba9bcf3203ffd5094a3ed631c083c/vaadin-radio-button.html#L144)):*
> ```HTML
> 
>           if (!this.value) {
>             this.value = this.innerText;
> ```

why we take the inner text as value, I think it is responsibility of the developer set a value to the control otherwise it should be on when checked, like in the native input radio.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks broke.

---

*[bower.json, line 18 at r12](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoGmGd7majdQtR6pNkr:-KoGmGd7majdQtR6pNks:bxaj0kw) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/88dfbcfd6cbba9bcf3203ffd5094a3ed631c083c/bower.json#L18)):*
> ```JSON
>   ],
>   "license": "Apache-2.0",
>   "ignore": [
> ```

add the `package-lock.json` to this file

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

manolo commented:
You need to fix the build in travis.

---

Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoGmeCpNIKgHN9f-nF0:b-3fez19)*
<!-- Sent from Reviewable.io -->

limonte commented:
Reviewed 1 of 19 files at r2, 1 of 1 files at r7, 1 of 2 files at r8, 4 of 4 files at r9.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks broke.

---

*[vaadin-radio-button-group.html, line 18 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCTqBZkX14k_gEeRBc:-KoGmeMjqvmPCdMmrOyY:b-pztooi) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L18)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Do we really need flex layout for the radio buttons? Why?
</blockquote></details>

I believe it's because of better vertical alignment. It's ugly by default in Win/Linux/Chrome.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Build is fixed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

Review status: 20 of 29 files reviewed at latest revision, 11 unresolved discussions.

---

*[bower.json, line 18 at r12](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoGmGd7majdQtR6pNkr:-KoHJIE66x24MmBALRNJ:bvgp7ap) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/88dfbcfd6cbba9bcf3203ffd5094a3ed631c083c/bower.json#L18)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

add the `package-lock.json` to this file
</blockquote></details>

Done in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button-group.html, line 18 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCTqBZkX14k_gEeRBc:-KoHJX1tudsrWqryDIFH:bj7q1sw) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L18)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

I believe it's because of better vertical alignment. It's ugly by default in Win/Linux/Chrome.
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43 , so there will be no problems with baseline. 

---

*[vaadin-radio-button-group.html, line 32 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCQMun5Oxo6bRVGhYK:-KoHJcUkPUvkzk-aRTbx:bkk5w64) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L32)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Is wrapper necessary here? Why?
</blockquote></details>

It was for future use with label. Removed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button-group.html, line 60 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCdkrTpoqby_nVDABI:-KoHK7siixY6bCEkACmD:b-gsez6f) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L60)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

This entire styling doc section is copied from checkbox, seems not relevant to the radio button group.

Let’s remove it for now. Another option is to rewrite it, so that it describes relevant stuff here.
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button-group.html, line 163 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCXgqOko-tca-8c8z7:-KoHKL4hIRPGhSpOA_nq:b-9o2l0p) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button-group.html#L163)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

probably specifying a closer selector '> vaadin-radio-button'  works without having to chain methods
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43 and now is using filtering from children.

---

*[vaadin-radio-button.html, line 36 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCcpL44gFJkLqqPuC3:-KoHKRvZaNvoJU92aPG4:bjhrqnz) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L36)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Afaik span should not have a native outline, no need for this like then. Please check if it is necessary.
</blockquote></details>

Checked and removed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button.html, line 85 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCeyEfE4AwPqaWf7e6:-KoHK_UyV_hLD97B1cpE:b-gsez6f) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L85)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

`checkbox` copy-pasting leftover here and below.
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button.html, line 151 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCixd4V6wvPalFvXaS:-KoHKfXBVNBfREGo_9Vh:b-mtz9vt) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/vaadin-radio-button.html#L151)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Hm-m, I’m not completely sure, but seems that we can simply call this event as `'change'`. 

Interference with the native `change` should not be an issue in this case. The original event from the internal native radio is not composed, therefore it should not propagate outside the shadow root.

Could you, please, try?
</blockquote></details>

Tested and changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[vaadin-radio-button.html, line 144 at r12](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoDBs1EXKqlqogNY607:-KoHKdBlidkh-vhv88TP:b-gsez6f) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/88dfbcfd6cbba9bcf3203ffd5094a3ed631c083c/vaadin-radio-button.html#L144)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

why we take the inner text as value, I think it is responsibility of the developer set a value to the control otherwise it should be on when checked, like in the native input radio.
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[test/vaadin-radio-button_test.html, line 24 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoCl90Tb6H4lL4SDpcW:-KoHKj4w5b6QDr_Y0ltO:bzhbxss) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/test/vaadin-radio-button_test.html#L24)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Seems like some tests are for the group, but some are just for the radio button.

We usually have separate test files for separate element files. Maybe split this up a bit?
</blockquote></details>

Separated in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---

*[test/vaadin-radio-button_test.html, line 31 at r11](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoClgROMCZe5yO9i_wB:-KoHKmhPbBAle1bAZNaT:b-gsez6f) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/2ad79089c34c43845e7f3af9100d77ba5b528c07/test/vaadin-radio-button_test.html#L31)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Using Polymer.dom is unnecessary with Polymer 2. You should be able to use the DOM APIs directly.
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1fb7747c0ecd740dbed341da90091a98e9cb1f43

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoHKqfsl6wz7XiFmVdH:b83oe2r)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 15 of 19 files at r2, 2 of 3 files at r5, 9 of 9 files at r13.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-radio-button.html, line 121 at r13](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoHb0it-iGsxAIJPnCA:-KoHb0it-iGsxAIJPnCB:b904u4) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1fb7747c0ecd740dbed341da90091a98e9cb1f43/vaadin-radio-button.html#L121)):*
> ```HTML
>             },
> 
>             disabled: {
> ```

Missing JSdoc description for the public property.

---

*[demo/group.html, line 31 at r13](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoHbC7O7ecA7EBGZDiu:-KoHbC7O7ecA7EBGZDiv:b-ki8bjf) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1fb7747c0ecd740dbed341da90091a98e9cb1f43/demo/group.html#L31)):*
> ```HTML
>     </demo-snippet>
> 
>     <h3>Radio group with iron form</h3>
> ```

Title case in the headings, please: “Radio Group with Iron Form”

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

YuriyVaadin commented:
Review status: 26 of 29 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-radio-button.html, line 121 at r13](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoHb0it-iGsxAIJPnCA:-KoHgcw_oSr5Ia5xfLlk:b-f0fbqk) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1fb7747c0ecd740dbed341da90091a98e9cb1f43/vaadin-radio-button.html#L121)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Missing JSdoc description for the public property.
</blockquote></details>

Done in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1c637b62cd746d325f564567aa59ecb9376c89ba

---

*[demo/group.html, line 31 at r13](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-KoHbC7O7ecA7EBGZDiu:-KoHgfDwysVAkRNB5oHS:b7rkjrg) ([raw file](https://github.com/vaadin/vaadin-radio-button/blob/1fb7747c0ecd740dbed341da90091a98e9cb1f43/demo/group.html#L31)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Title case in the headings, please: “Radio Group with Iron Form”
</blockquote></details>

Changed in https://github.com/vaadin/vaadin-radio-button/pull/2/commits/1c637b62cd746d325f564567aa59ecb9376c89ba

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2)*
<!-- Sent from Reviewable.io -->

platosha commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Well done!

---

Reviewed 1 of 19 files at r2, 3 of 3 files at r14.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoHiKYAulXrt6PMsS1r:b-wehp9g)*
<!-- Sent from Reviewable.io -->

limonte commented:
Great job indeed! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 1 of 19 files at r2, 1 of 2 files at r11, 1 of 4 files at r12, 9 of 9 files at r13.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoHdAxKFyNII6k-3wWN:bfr5zqt)*
<!-- Sent from Reviewable.io -->

manolo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> congrats, your first element!

---

Reviewed 15 of 19 files at r2, 2 of 3 files at r5, 1 of 1 files at r7, 1 of 4 files at r9, 1 of 2 files at r11, 1 of 4 files at r12, 6 of 9 files at r13, 3 of 3 files at r14.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-radio-button/2#-:-KoHqmz4NmUQ2Q02lLBD:b-518mxg)*
<!-- Sent from Reviewable.io -->
